### PR TITLE
Update date-time-functions.md

### LINKED ISSUE #42758: toYYYYMMDDhhmmss(): second argument is not documented.
```
SELECT toYYYYMMDDhhmmss(now(), 2)

Query id: 4d7a3005-4d57-4e40-9321-d80e02f0621d


0 rows in set. Elapsed: 0.001 sec. 

Received exception from server (version 22.3.2):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Function toYYYYMMDDhhmmss supports 1 or 2 arguments. The 1st argument must be of type Date or DateTime. The 2nd argument (optional) must be a constant string with timezone name: While processing toYYYYMMDDhhmmss(now(), 2). (ILLEGAL_TYPE_OF_ARGUMENT)
```

https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions/#toyyyymmddhhmmss:

image

Possibly other related functions.

Please document.

### PR DESCRIPTION
**Pull request by DanRoscigno**
closes #42758

Add optional timezone arg to tyYYYYMMDD etc.

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 1]
```diff
@@ -1126,15 +1126,48 @@ Rounds the time to the half hour.
...
+```response
+┌─toYYYYMMDD(now(), 'US/Eastern')─┐
+│                        20230302 │
+└─────────────────────────────────┘
+```
 
 ## toYYYYMMDDhhmmss
 
-Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss).
+Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss).  Accepts a second optional timezone argument.  If provided, the timezone must be a string constant.
```
Comment (qoega) [inline docs/en/sql-reference/functions/date-time-functions.md:1159]:
```suggestion
Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss).  Accepts a second optional timezone argument. If provided, the timezone must be a string constant.
```

Comment (qoega) [reply docs/en/sql-reference/functions/date-time-functions.md:1159]: Or it is intended in all cases?

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 2]
```diff
@@ -1126,15 +1126,48 @@ Rounds the time to the half hour.
 
 ## toYYYYMM
 
-Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 100 + MM).
+Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 100 + MM). Accepts a second optional timezone argument.  If provided, the timezone must be a string constant.
```
Comment (qoega) [inline docs/en/sql-reference/functions/date-time-functions.md:1129]:
```suggestion
Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 100 + MM). Accepts a second optional timezone argument. If provided, the timezone must be a string constant.
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 3]
```diff
@@ -1126,15 +1126,48 @@ Rounds the time to the half hour.
...
+```response
+┌─toYYYYMM(now(), 'US/Eastern')─┐
+│                        202303 │ 
+└───────────────────────────────┘
+```
 
 ## toYYYYMMDD
 
-Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 10000 + MM \* 100 + DD).
+Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 10000 + MM \* 100 + DD).  Accepts a second optional timezone argument.  If provided, the timezone must be a string constant.
```
Comment (qoega) [inline docs/en/sql-reference/functions/date-time-functions.md:1144]:
```suggestion
Converts a date or date with time to a UInt32 number containing the year and month number (YYYY \* 10000 + MM \* 100 + DD). Accepts a second optional timezone argument. If provided, the timezone must be a string constant.
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 4]
```diff
@@ -1126,15 +1126,48 @@ Rounds the time to the half hour.
...
+```response
+┌─toYYYYMMDD(now(), 'US/Eastern')─┐
+│                        20230302 │
+└─────────────────────────────────┘
+```
 
 ## toYYYYMMDDhhmmss
 
-Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss).
+Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss).  Accepts a second optional timezone argument. If provided, the timezone must be a string constant.
```
Comment (qoega) [inline docs/en/sql-reference/functions/date-time-functions.md:1159]:
```suggestion
Converts a date or date with time to a UInt64 number containing the year and month number (YYYY \* 10000000000 + MM \* 100000000 + DD \* 1000000 + hh \* 10000 + mm \* 100 + ss). Accepts a second optional timezone argument. If provided, the timezone must be a string constant.
```

**Review by qoega — APPROVED** [review]
Then all other double whitespaces also :)

**Comment by DanRoscigno** [comment]
> Then all other double whitespaces also :)

It does not get rendered, it is just proper writing (or at least it used to be).  My fingers do it automatically.

**Pull request merged**